src/components/StackNavigation/Dots/StackNavigationDots.tsx

```diff
@@ -0,0 +1,54 @@
+import classNames from "classnames";
+import {useNavigate} from "react-router";
+import {Note} from "types/note";
+import "./StackNavigationDots.scss";
+
+type StackNavigationDotsProps = {
+  stacks: Note[];
+  currentIndex: number;
+  handleModeration: (stackId: string) => void;
+};
+
+const getCenterDot = (length: number, currentIndex: number) => {
+  if (currentIndex < 3) {
+    return 3;
+  }
+  if (length - currentIndex < 4) {
+    return length - 4;
+  }
+  return currentIndex;
+};
+
+export const StackNavigationDots = ({stacks, currentIndex, handleModeration}: StackNavigationDotsProps) => {
+  const navigate = useNavigate();
+
+  const handleClick = (index: number) => {
+    handleModeration(stacks[index].id);
+    navigate(`../note/${stacks[index].id}/stack`);
+  };
+
+  const centerDot = getCenterDot(stacks.length, currentIndex);
+
+  return (
+    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+    <div className="stack-view__navigation-dots" onClick={(e) => e.stopPropagation()}>
```

Comment:
it's still needed because otherwise the portal would close every time one of the dots is pressed. I could move it to the buttons now but I don't see a benefit in that